src/main/java/com/avairebot/commands/fun/UndertaleTextBoxCommand.java

```diff
@@ -0,0 +1,99 @@
+package com.avairebot.commands.fun;
+
+import com.avairebot.AvaIre;
+import com.avairebot.commands.CommandMessage;
+import com.avairebot.contracts.commands.Command;
+import com.avairebot.language.I18n;
+import net.dv8tion.jda.core.EmbedBuilder;
+import net.dv8tion.jda.core.MessageBuilder;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+public class UndertaleTextBoxCommand extends Command {
+
+    private final String templateUrl = "https://www.demirramon.com/gen/undertale_box.png?character={0}&message={1}&ext=.png";
+
+    public UndertaleTextBoxCommand(AvaIre avaire) {
+        super(avaire);
+    }
+
+    @Override
+    public String getName() {
+        return "Undertale TextBox Command";
+    }
+
+    @Override
+    public String getDescription() {
+        return "TODO";
+    }
+
+    @Override
+    public List<String> getUsageInstructions() {
+        return Collections.singletonList(
+            "`:command <character> <message>` - Returns an undertale textbox image containing the specified text."
+        );
+    }
+
+    @Override
+    public List<String> getExampleUsage() {
+        return Collections.singletonList("`:command toriel Greetings, my child`");
+    }
+
+    @Override
+    public List<String> getMiddleware() {
+        return Collections.singletonList("throttle:channel,2,5");
+    }
+
+    @Override
+    public List<String> getTriggers() {
+        return Arrays.asList("utbox", "textbox");
+    }
+
+    @Override
+    public boolean onCommand(CommandMessage context, String[] args) {
+        if (args.length == 0) {
+            return sendErrorMessage(context, "errors.missingArgument", "character");
```

Comment:
The characters list could be displayed here instead of the error message, could store the character names in a string list, and then use the [SimplePaginator](https://github.com/avaire/avaire/blob/master/src/main/java/com/avairebot/chat/SimplePaginator.java) class to show the character names 10 at a time over multiple pages.

Reply:
I allowed custom character images to be displayed through the author's URL field. I also allowed users to not have to specify a character when creating a text bot.